Fix apollo-server null issue due to missing columns

zachrdz opened the pull request:
By submitting a PR to this repository, you agree to the terms within the [Code of Conduct](https://github.com/join-monster/join-monster/blob/master/CODE_OF_CONDUCT.md). Please see the [contributing guidelines](https://github.com/join-monster/join-monster/blob/master/CONTRIBUTING.md) for how to create and submit a high-quality PR for this repo.

### Description

Recent apollo-server-core changes always define a field resolver (see enablePluginsForSchemaResolvers function, apollo-server issue #3988) so '!field.resolve' is not a good check for columns. Because of this, the query is not built correctly and does not included the requested columns, leading to all fields being null when using `ApolloServer`. Instead use parentTypeNode.constructor.name.

Current condition
```
if (fieldConfig.sqlColumn || !field.resolve)
```

Proposed condition
```
if (fieldConfig.sqlColumn || ['GraphQLObjectType', 'GraphQLInterfaceType'].includes(parentTypeNode.constructor.name))
```

### References

[Reference to existing PR in draft state](https://github.com/join-monster/join-monster/pull/436)

[Apollo Server breaking change](https://github.com/apollographql/apollo-server/pull/3988)

### Testing

Modified an existing test for `User > favNums` resolver to add `ignoreAll` since it does not use join-monster functionality.

- [x] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [x] I have added documentation for new/changed functionality in this PR via comments and by updating the change log
- [x] All active GitHub checks for tests, formatting, and security are passing

alexbbt commented on package.json:
```diff
@@ -121,4 +121,4 @@
     "lodash": "^4.17.15"
   }
-}
+}
```
Can you put the new line back?

zachrdz replied:
Thanks! Yes, I've just committed the change to flatten my commits and remove this extra line.

alexbbt reviewed (changes requested):
Looks good, I will merge this, just one thing to fix

zachrdz committed:
Fix apollo-server null issue due to missing columns

Recent apollo-server-core changes always define a field resolver (see the enablePluginsForSchemaResolvers function, apollo-server issue #3988) so '!field.resolve' is not a good check for columns. Instead use parentTypeNode.constructor.name.

Incremented release version and updated change log.
